src/ensembl/src/content/app/entity-viewer/gene-view/components/collapsed-exons-image/CollapsedExonsImage.tsx

```diff
@@ -76,28 +71,22 @@ export const ExonsImage = (props: ExonsImageProps) => {
 const ExonsImageWithData = (props: ExonsImageWithDataProps) => {
   const { transcript } = props;
 
-  if (!transcript.cds) {
+  if (!transcript.product) {
     return null;
   }
 
-  const codingExons = getCodingExonsForImage(transcript);
-
-  const getSplicedRNALength = () => {
-    const rnaLength = transcript.exons.reduce((length, exon) => {
-      const { start, end } = getFeatureCoordinates(exon);
-      return length + (end - start + 1);
-    }, 0);
-
-    return getCommaSeparatedNumber(rnaLength);
-  };
-
+  // If we consider the image's starting point as A and end point as B the scale can be sketched as:
+  // A--  longest protein's length  --B
+  // A--      width in pixels       --B
   const scale = scaleLinear()
-    .domain([0, props.refCDSLength])
+    .domain([0, props.longestProteinLength])
     .range([0, props.width])
     .clamp(true);
 
   const midStyles = classNames(transcriptsListStyles.middle, styles.middle);
 
+  // The scale/longest protein is displayed using the first <rect> from point A to point B
+  // The protein is displayed from point A (check comment above) using the second <rect>
```

Comment:
> The scale/longest protein is displayed

Didn't we agree that we call this light-grey rectangle over which we draw darker (or blue) rectangles a track (or a lane, whatever)? Calling it "the longes protein" in this comment is confusing.

Also, it should be fairly obvious what each rectangle represents from the names of their classes. Is this comment adding anything useful?

Reply:
> Also, it should be fairly obvious what each rectangle represents from the names of their classes. Is this comment adding anything useful?

It probably doesn't anymore, since the comment will be changed for the scale code. I'll remove this.

Reply:
✔️